pkg/apimachinery/identity/wrapper.go

```diff
@@ -13,6 +14,37 @@ type IDClaimsWrapper struct {
 	Source Requester
 }
 
+func (i *IDClaimsWrapper) NamespaceMatches(namespace string) bool {
+	parts := strings.Split(i.Source.GetAllowedKubernetesNamespace(), "-")
+	// for multi-tenant operators, access tokens will have a namespace claim of "*"
+	if parts[0] == "*" {
+		return true
+	}
+
+	if len(parts) < 2 {
+		return false
+	}
+
+	// for cluster-scoped resources, namespace in request is ""
+	if namespace == "" {
+		return true
+	}
+
+	namespaceParts := strings.Split(namespace, "-")
+	if len(namespaceParts) < 2 {
+		return false
+	}
+	if parts[0] == "stack" || parts[0] == "stacks" {
+		return namespaceParts[1] == parts[1]
+	}
+
+	return false
```

Comment:
is this needed to support `org-*`?
```suggestion
	return i.Source.GetAllowedKubernetesNamespace() == namespace
```

Reply:
Yes, i will accept your suggestion. The Namespace matches isn't used anywhere so far, just fyi, except in my enterprise PR for the standalone authz.